retriever/postgresqlretriever/retriever.go

```diff
@@ -0,0 +1,119 @@
+package postgresqlretriever
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/thomaspoignant/go-feature-flag/retriever"
+	"github.com/thomaspoignant/go-feature-flag/utils/fflog"
+)
+
+// Retriever is a configuration struct for a PostgreSQL connection
+type Retriever struct {
+	// PostgreSQL connection URI
+	URI  string
+	Type string
+	// PostgreSQL table where flag column is stored
+	Table string
+	// PostgreSQL column where flag definitions are stored
+	Column string
+	dbPool *pgxpool.Pool
+	status string
+	logger *fflog.FFLogger
+}
+
+func (r *Retriever) Init(ctx context.Context, logger *fflog.FFLogger) error {
+	r.logger = logger
+	if r.dbPool == nil {
+		r.status = retriever.RetrieverNotReady
+
+		poolConfig, err := pgxpool.ParseConfig(r.URI)
```

Comment:
**❔ question:** ‏Do we need a connection pool here? We will have only 1 connection per retriever, I am not super familiar with `pgxpool` but what value is it bringing here?

Reply:
You are right. Changed it to direct connection instead of a pool. If we have single connection it won't bring any value.